app/src/main/java/com/example/albumsapp/viewmodel/PhotoViewModel.kt

```diff
@@ -1,25 +1,31 @@
 package com.example.albumsapp.viewmodel
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import androidx.lifecycle.ViewModel
-import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.*
 import com.example.albumsapp.model.Photo
 import com.example.albumsapp.network.AlbumsApi
 import kotlinx.coroutines.launch
 
-class PhotoViewModel : ViewModel() {
+class PhotoViewModel(val id: String) : ViewModel() {
     private val _photos = MutableLiveData<List<Photo>>()
 
     val photos: LiveData<List<Photo>> = _photos
 
+    class MyViewModelFactory(
+        private val id: String
+    ) : ViewModelProvider.NewInstanceFactory() {
+        override fun <T : ViewModel> create(modelClass: Class<T>): T {
+            return PhotoViewModel(id) as T
+        }
+    }
+
     init {
+        getPhotos(id.toInt())
     }
 
-    private fun getPhotos(index: String) {
+    private fun getPhotos(albumId: Int) {
         viewModelScope.launch {
             try {
-                _photos.value = AlbumsApi.retrofitService.getPhotos(index)
+                _photos.value = AlbumsApi.retrofitService.getPhotos(albumId)
             } catch (e: Exception) {
             }
```

Comment:
let's not ignore exceptions. We have to show error somehow to user.
Let's try to use `AlertDialog ` to show an error.

Reply:
Can we used AlertDialog in viewModel? Because as i know ViewModel used for store data

Reply:
yes, you are right.  Usually we make a separate error live data (LiveData\<Exception\>) and observe it in fragment and then  show  AlertDialog there.
in `catch` we just do `errorLiveData.postValue(e)`

Reply:
How i can handle exception from viewmodel in fragment. I should use errorLiveData for this?

Reply:
yes, you have to observe `errorLiveData` in the same manner you observe `photos` livedata